src/neo/SmartContract/InteropService.Storage.cs

```diff
@@ -24,9 +25,28 @@ public static class Storage
             public static readonly InteropDescriptor PutEx = Register("System.Storage.PutEx", Storage_PutEx, GetStoragePrice, TriggerType.Application, CallFlags.AllowModifyStates);
             public static readonly InteropDescriptor Delete = Register("System.Storage.Delete", Storage_Delete, 0_01000000, TriggerType.Application, CallFlags.AllowModifyStates);
 
-            private static long GetStoragePrice(EvaluationStack stack)
+            private static long GetStoragePrice(EvaluationStack stack, StoreView snapshot)
             {
-                return (stack.Peek(1).GetByteLength() + stack.Peek(2).GetByteLength()) * GasPerByte;
+                var key = stack.Peek(1);
+                var value = stack.Peek(2);
+                var newDataSize = value.IsNull ? 0 : value.GetByteLength();
+                if (!(stack.Peek() is InteropInterface _interface))
+                    throw new InvalidOperationException();
+
+                StorageContext context = _interface.GetInterface<StorageContext>();
+                StorageKey skey = new StorageKey
+                {
+                    Id = context.Id,
+                    Key = key.GetSpan().ToArray()
+                };
+                var skeyValue = snapshot.Storages.TryGet(skey);
+                if (skeyValue is null)
+                    newDataSize += key.GetByteLength();
+                else if (newDataSize <= skeyValue.Value.Length)
+                    newDataSize = 1;
```

Comment:
I think the minimum cost should be the same as `Delete`.

Reply:
Reasonable, but why not set `Delete` cost to `1 * GasPerByte`. 
Currently the cost of `Delete`  is equal to `10 * GetPerByte`.

Reply:
> Reasonable, but why not set Delete cost to 1 * GasPerByte.

It's okay to me.